langchain/chains/conversation/memory.py

```diff
@@ -40,6 +46,88 @@ def _save_context(self, inputs: Dict[str, Any], outputs: Dict[str, str]) -> None
         self.buffer += "\n" + "\n".join([human, ai])
 
 
+class ConversationEntityMemory(Memory, BaseModel):
+    """Conversation entity search and store to memory."""
+
+    buffer: List[dict] = Field(default_factory=list)
+    docstore: InMemoryDocstore
+    llm: LLM
+    entity_extraction_prompt: BasePromptTemplate = ENTITY_EXTRACTION_PROMPT
+    entity_summarization_prompt: BasePromptTemplate = ENTITY_SUMMARIZATION_PROMPT
+    history_window_len: int = 5  #: :meta private:
+    dynamic_key_list: List[str] = ["history", "entities"]  #: :meta private:
+    entities_per_turn: List[str] = Field(default_factory=list)  #: :meta private:
+
+    @property
+    def dynamic_keys(self) -> List[str]:
+        """Will always return list of dynamic keys.
+
+        :meta private:
+        """
+        return self.dynamic_key_list
+
+    def _format_buffer_to_string(self) -> str:
+        section = self.buffer[-self.history_window_len :]
```

Comment:
will we really want this on the other memory objects as well?

Reply:
if so, i wonder if theres a way to use two memory objects (as far as i cn tell, the history and entity logic is pretty unrelated?